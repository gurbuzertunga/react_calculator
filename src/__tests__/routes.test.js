import renderer from 'react-test-renderer';
import Route from '../components/routes';
import MathMagicians from '../pages/mathmagicians';
import Quote from '../pages/quote';
import Calculator from '../pages/calculator';

describe('checks if the Routes have the correct directories', () => {
  test('should render to Home route', () => {
    const component = MathMagicians;

    const tree = renderer
      .create(<Route path="/" component={component} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render to App route', () => {
    const component = Calculator;

    const tree = renderer
      .create(<Route path="/calculator" component={component} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render to Quote route', () => {
    const component = Quote;

    const tree = renderer
      .create(<Route path="/calculator" component={component} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
