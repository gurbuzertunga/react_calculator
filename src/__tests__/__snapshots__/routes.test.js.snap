// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if the Routes have the correct directories should render to App route 1`] = `
Array [
  <header>
    <h1>
      Math Magicians
    </h1>
    <nav>
      <li
        className="link"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="link"
      >
        <a
          href="/calculate"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="link"
      >
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </nav>
  </header>,
  <div
    className="content"
  >
    <h2>
      Welcome to our page!
    </h2>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Commodi voluptatum consectetur eligendi praesentium, corporis quibusdam ipsa, aliquam facere veniam earum atque iusto sint tempora? Amet corrupti provident adipisci rerum inventore.
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Molestiae, aliquid, nesciunt deleniti unde, earum iure sapiente neque fugit aliquam corrupti quaerat incidunt totam nostrum. Nisi sequi reiciendis facere quos ut?
    </p>
  </div>,
]
`;

exports[`checks if the Routes have the correct directories should render to Home route 1`] = `
Array [
  <header>
    <h1>
      Math Magicians
    </h1>
    <nav>
      <li
        className="link"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="link"
      >
        <a
          href="/calculate"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="link"
      >
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </nav>
  </header>,
  <div
    className="content"
  >
    <h2>
      Welcome to our page!
    </h2>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Commodi voluptatum consectetur eligendi praesentium, corporis quibusdam ipsa, aliquam facere veniam earum atque iusto sint tempora? Amet corrupti provident adipisci rerum inventore.
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Molestiae, aliquid, nesciunt deleniti unde, earum iure sapiente neque fugit aliquam corrupti quaerat incidunt totam nostrum. Nisi sequi reiciendis facere quos ut?
    </p>
  </div>,
]
`;

exports[`checks if the Routes have the correct directories should render to Quote route 1`] = `
Array [
  <header>
    <h1>
      Math Magicians
    </h1>
    <nav>
      <li
        className="link"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="link"
      >
        <a
          href="/calculate"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="link"
      >
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </nav>
  </header>,
  <div
    className="content"
  >
    <h2>
      Welcome to our page!
    </h2>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Commodi voluptatum consectetur eligendi praesentium, corporis quibusdam ipsa, aliquam facere veniam earum atque iusto sint tempora? Amet corrupti provident adipisci rerum inventore.
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Molestiae, aliquid, nesciunt deleniti unde, earum iure sapiente neque fugit aliquam corrupti quaerat incidunt totam nostrum. Nisi sequi reiciendis facere quos ut?
    </p>
  </div>,
]
`;
